apps/server/src/app/structure-definition.controller.ts

```diff
@@ -99,28 +101,38 @@ export class StructureDefinitionController extends BaseFhirController {
    * @param type {string}
    */
   @Get('base')
-  public async getBaseStructureDefinition(@Req() request: ITofRequest, @Query('url') url: string, @Query('type') type: string): Promise<BaseDefinitionResponseModel> {
+  public async getBaseStructureDefinition(@Req() request: ITofRequest, @Query('url') url: string, @Query('type') type: string, @RequestHeaders("implementationGuideId") implementationGuideId: string,
+                                          @FhirServerVersion() fhirServerVersion: 'stu3'|'r4'): Promise<BaseDefinitionResponseModel> {
     const ret = new BaseDefinitionResponseModel();
     const fhirServer = this.configService.fhir.servers.find(s => s.id === request.fhirServerId);
 
     if (!url.startsWith('http://hl7.org/fhir/StructureDefinition/')) {
       try {
-        let profileWithSnapshot: IStructureDefinition = await this.getBaseStructureDefinitionResource(request.fhirServerBase, url);
-
-        // The snapshot is not already generated for the profile, so we need to generate it now.
-        if (!profileWithSnapshot.snapshot) {
-          if (fhirServer.supportsSnapshot) {
-            const snapshotUrl = buildUrl(request.fhirServerBase, 'StructureDefinition', null, '$snapshot', { url: url });
-            const results = await this.httpService.get(snapshotUrl).toPromise();
-            profileWithSnapshot = results.data;
-          } else {
-            profileWithSnapshot = await this.generateInternalSnapshot(request.fhirServerBase, request.fhir, url);
-          }
-        }
+        const dependencies = await this.fhirController.searchDependency(request.fhirServerBase, implementationGuideId, fhirServerVersion, 'StructureDefinition',
+          null, null, null, null, null, url, false);
 
-        if (profileWithSnapshot && profileWithSnapshot.resourceType === 'StructureDefinition' && profileWithSnapshot.snapshot) {
-          ret.base = profileWithSnapshot;
+        if (dependencies.entry.length === 1) {
+          ret.base = dependencies.entry[0].resource['resource'];
           return ret;
+        } else {
```

Comment:
because `return ret;` is in the line above, I don't think you need the `} else { ... }`
you can just call the code _in_ the `else` after the `if (dependencies.entry.length === 1) { ... }` block because if there _is_  a single dependency, then it will stop the code from progressing
the code works either way, it is just my preference not to use `else` when it is not needed...

Reply:
very true, good point, thank you